### DISCUSSION TITLE
Debugging using VS Code

### DISCUSSION THREAD

**Discussion by JoobyPM** [nitrojs/nitro#2057] [Q&A] [answered]
Hello, dear Nitro community!

I'm new to using TypeScript (TS) for Node.js development, and I'm particularly interested in Nitro. I'm eager to transition to using Nitro with TS, but I'm struggling with how to set up the debugging tool.

Likewise, I followed the initial steps in the documentation:

```bash
pnpm dlx giget@latest nitro nitro-app
```

However, when I tried to configure the debugger in VSCode, I couldn't get it to work. Could anyone kindly assist me with this?

Would it be possible for someone to share an example `.vscode/launch.json` configuration? It would be really helpful!

Thank you in advance for your help!

**Comment by JoobyPM** [comment, marked as answer]
Thanks to ChatGPT, I managed to configure the VS Code debugger.
`.vscode/launch.json`:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Debug Nitro",
            "runtimeExecutable": "${workspaceFolder}/node_modules/.bin/nitropack",
            "runtimeArgs": [
                "dev"
            ],
            "port": 9229,
            "outputCapture": "std",
            "restart": true,
            "console": "integratedTerminal",
            "skipFiles": [
                "<node_internals>/**"
            ]
        }
    ]
}
```

It would be great to add "how to debug" into [Getting Started](https://nitro.unjs.io/guide/getting-started)

**Comment by Harm-Nullix** [comment]
For everyone landing on this apge with Jetbrains, I just made a discussion about this for Nitro.
You can check it out [here](https://github.com/unjs/nitro/discussions/2141)
